src/Slawek/Fsd.Slawek.Cs.Ex4/Fsd.Slawek.Cs.Ex4.Web/Controllers/SingleProductController.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Fsd.Slawek.Cs.Ex4.Data.Entities;
+using Fsd.Slawek.Cs.Ex4.Services.Services.Product;
+using Fsd.Slawek.Cs.Ex4.Web.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+
+namespace Fsd.Slawek.Cs.Ex4.Web.Controllers
+{
+    public class SingleProductController : Controller
+    {
+        IProductService _productService;
+
+        public SingleProductController()
+        {
+            _productService = new MemoryProductService();
+        }
+
+        public ActionResult Index(int index = 0)
+        {
+            var product = new SingleProduct<ProductBase>();
+
+            product = product.GetInfo(_productService.GetAllProducts().ToList()[index]);
```

Comment:
takie pobieranie będzie bardzo niewydajne - najpierw pobierasz całą listę a następnie wyciągasz jeden element. Jak podmienimy to na bazę danych to będzie problem. Przygotuj odpowiedni interfejs na poziomie serwisu aby wyciągać jeden element na podstawie indexu/id itp